site/js/modules/webui/server.js

```diff
@@ -0,0 +1,190 @@
+        })
+    ;
+}
+
+function render(endpoints, selectedEndpoint, context, container) {
+    let selector = renderSelector(endpoints, selectedEndpoint);
+    let endpointArea = renderEndpointArea(endpoints, selectedEndpoint, context);
+
+    let html = `
+        <div class="page-controls">${selector}</div>
+        <div class="endpoint-area">${endpointArea}</div>
+        <div class="results-area"></div>
+    `;
+
+    container.innerHTML = html;
+
+    container.querySelector(".page-controls select").addEventListener("change", function(event) {
+        let newParams = {
+            [Routing.PARAM_TARGET_ENDPOINT]: event.target.value,
+        };
+
+        let path = Routing.formHashPath(Routing.PATH_SERVER_CALL_API, newParams);
+        Routing.redirect(path);
+    });
+
+    let button = container.querySelector(".endpoint-area button");
+
+    if (button) {
+        button.addEventListener("click", function(event) {
+            callEndpoint(selectedEndpoint, endpoints[selectedEndpoint]["input"], context, container);
+        });
+    }
+}
+
+function renderSelector(endpoints, selectedEndpoint) {
+    let optionsList = [];
+
+    for (const endpoint of Object.keys(endpoints)) {
+        let isSelected = "";
+        if (endpoint === selectedEndpoint) {
+            isSelected = "selected";
+        }
+
+        optionsList.push(`<option value="${endpoint}" ${isSelected}>${endpoint}</option>`);
+    }
+
+    return `
+        <select id="endpoint-dropdown">
+            <option value="">Select an endpoint...</option>
+            ${optionsList.join("\n")}
+        </select>
+    `;
+}
+
+function renderEndpointArea(endpoints, selectedEndpoint, context) {
+    if (!(selectedEndpoint in endpoints)) {
+        return '';
+    }
+
+    let sortedInputs = endpoints[selectedEndpoint]["input"];
+    sortedInputs.sort(function(a, b) {
+        let aPriority = fieldPriority.indexOf(a.name);
+        let bPriority = fieldPriority.indexOf(b.name);
+
+        return bPriority - aPriority;
+    });
+
+    let inputFields = [];
+    for (const field of sortedInputs) {
+        let inputType = "text";
+        let placeholder = "";
+
+        if (field.name === "user-email") {
+            placeholder = context.user.email;
+        } else if (field.name === "user-pass") {
+            placeholder = "<current token>";
+            inputType = "password";
+        } else if (field.type.includes("SelfOr")) {
+            placeholder = context.user.email;
+        }
+
+        inputFields.push(`
+            <div class="input-field">
+                <label for="${field.name}">${field.name} (expects: ${field.type})</label>
+                <input type="${inputType}" id="${field.name}" name="${field.name}" placeholder="${placeholder}">
+            </div>
+        `);
+    }
+
+    return `
+        <h2>
+            ${selectedEndpoint}
+        </h2>
+
+        <fieldset>
+            ${inputFields.join("\n")}
+        </fieldset>
```

Comment:
What do you think about putting an event listener on this (keydown probably) to listen for enter/return and calling on that?
(Note that it would not be added directly in the HTML, but by querying for it after.)